addons/binding/org.openhab.binding.lutron/src/main/java/org/openhab/binding/lutron/internal/hw/HwSerialBridgeHandler.java

```diff
@@ -0,0 +1,275 @@
+
+        // For good measure, we'll call it once now.
+        updateProcessorTime();
+    }
+
+    private void updateProcessorTime() {
+        Date date = new Date();
+        String dateString = dateFormat.format(date.getTime());
+        String timeString = timeFormat.format(date.getTime());
+        logger.debug("Updating HomeWorks processor date and time to {} {}", dateString, timeString);
+
+        if (!this.getBridge().getStatus().equals(ThingStatus.ONLINE)) {
+            logger.warn("HomeWorks Bridge is offline and cannot update time on HomeWorks processor.");
+            if (updateTimeJob != null) {
+                updateTimeJob.cancel(false);
+                updateTimeJob = null;
+            }
+            return;
+        }
+
+        sendCommand("SD, " + dateString);
+        sendCommand("ST, " + timeString);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.error("Unexpected command for HomeWorks Bridge: {} - {}", channelUID, command);
+    }
+
+    private void handleIncomingMessage(String line) {
+        if (line == null || line.isEmpty()) {
+            return;
+        }
+
+        logger.info("Received message from HomeWorks processor: {}", line);
+        String[] data = line.replaceAll("\\s", "").toUpperCase().split(",");
+        if ("DL".equals(data[0])) {
+            try {
+                String address = data[1];
+                Integer level = Integer.parseInt(data[2]);
+                HwDimmerHandler handler = findHandler(address);
+                if (handler == null) {
+                    discService.declareUnknownDimmer(address);
+                } else {
+                    handler.handleLevelChange(level);
+                }
+            } catch (final Exception e) {
+                logger.error("Error parsing incoming message", e);
+            }
+        }
+
+    }
+
+    private HwDimmerHandler findHandler(String address) {
+        for (Thing thing : getThing().getThings()) {
+            if (thing.getHandler() instanceof HwDimmerHandler) {
+                HwDimmerHandler handler = (HwDimmerHandler) thing.getHandler();
+                if (address.equals(handler.getAddress())) {
+                    return handler;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Receives Serial Port Events and reads Serial Port Data.
+     *
+     * @param serialPortEvent
+     */
+    @Override
+    public void serialEvent(SerialPortEvent serialPortEvent) {
+        if (serialPortEvent.getEventType() == SerialPortEvent.DATA_AVAILABLE) {
+            try {
+                while (true) {
+                    String messageLine = serialInput.readLine();
+                    if (messageLine == null) {
+                        break;
+                    }
+                    handleIncomingMessage(messageLine);
+                }
+            } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception